src/battle_util.c

```diff
@@ -8820,6 +8837,8 @@ u16 CalcTypeEffectivenessMultiplier(u16 move, u8 moveType, u8 battlerAtk, u8 bat
         modifier = CalcTypeEffectivenessMultiplierInternal(move, moveType, battlerAtk, battlerDef, recordAbilities, modifier);
         if (gBattleMoves[move].effect == EFFECT_TWO_TYPED_MOVE)
             modifier = CalcTypeEffectivenessMultiplierInternal(move, gBattleMoves[move].argument, battlerAtk, battlerDef, recordAbilities, modifier);
+        else if (moveType == TYPE_FIRE && gDisableStructs[battlerDef].tarShot)
```

Comment:
would this be better in `MulByTypeEffectiveness`?

Reply:
Yeah that would be better!